src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/StructFlyweightGenerator.java

```diff
@@ -685,9 +912,16 @@ private MemberFieldGenerator(
             public MemberFieldGenerator addMember(
                 String name,
                 TypeName type,
-                TypeName unsignedType)
+                TypeName unsignedType,
+                boolean usedAsSize)
             {
-                if (!type.isPrimitive())
+                if (usedAsSize)
+                {
+                    TypeName publicType = unsignedType != null ? unsignedType : type;
```

Comment:
Suggest renaming this as `effectiveType` or `generateType` since the intent is to use `unsignedType` for the effective type in preference to the `type` for generation.

Use of the term `publicType` is ambiguous because the type is most "publicly" defined in the IDL rather than the generated API.

Reply:
done (and in other similar places)